File: java/src/com/yhsif/onepwd/QSTileService.java
package com.yhsif.onepwd;

import android.content.Intent;
import android.service.quicksettings.TileService;

public class QSTileService extends TileService {
  private static boolean added = false;

  @Override
  public void onClick() {
    startActivityAndCollapse(new Intent(this, OnePwd.class));
  }

  @Override
  public void onTileAdded() {
    added = true;
    NotificationService.stop(this);
  }

  @Override
  public void onTileRemoved() {
    added = false;
    NotificationService.run(this);
  }

  static boolean added() {
    return added;
  }
}

File: src/com/yhsif/onepwd/QSTileService.kt
package com.yhsif.onepwd

import android.content.Intent
import android.service.quicksettings.TileService
import androidx.core.service.quicksettings.PendingIntentActivityWrapper
import androidx.core.service.quicksettings.TileServiceCompat

class QSTileService : TileService() {
  override fun onClick() {
    unlockAndRun() {
      val intent = Intent(this, OnePwd::class.java)
      intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
      TileServiceCompat.startActivityAndCollapse(
        this,
        PendingIntentActivityWrapper(
          this,
          0, // requestCode
          intent,
          Intent.FLAG_ACTIVITY_NEW_TASK,
          false, // isMutable
        ),
      )
    }
  }
}
